crates/loader/src/validation.rs

```diff
@@ -0,0 +1,47 @@
+use anyhow::{anyhow, Result};
+
+const ONLY_ALLOWED_KV_STORE: &str = "default";
+
+pub(crate) fn validate_key_value_stores(key_value_stores: &Option<Vec<String>>) -> Result<()> {
+    match key_value_stores {
+        None => Ok(()),
+        Some(ids) => match ids.iter().find(|id| *id != ONLY_ALLOWED_KV_STORE) {
```

Comment:
Another option here:

```suggestion
    match key_value_stores.iter().flatten().find(|id| *id != ONLY_ALLOWED_KV_STORE) {
```